src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Newtonsoft.Json.Linq;
+using System.Reflection;
+using System.Collections.ObjectModel;
+using System.IO;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    // A general purpose binder that pulls any self-describing SDK extensions. 
+    [CLSCompliant(false)]
+    public class GeneralScriptBindingProvider : ScriptBindingProvider    
+    {
+        private ToolingHelper _helper;
```

Comment:
ToolingHelper (needs better name) is a new type from SDK that traverses the rules in the extensions and helps us answer questions. This is the magic.  Having a single class here exposes a small surface area - rather than exposing all the different rules.